src/components/UseV2ExchangeModal/index.tsx

```diff
@@ -29,42 +29,46 @@ const StyledButton = styled(Button)`
 
 type UseV2ExchangeModalProps = {
   onDismiss?: () => void
+  isPageVisit?: boolean
 }
 
-const UseV2ExchangeModal = ({ onDismiss = defaultOnDismiss }: UseV2ExchangeModalProps) => {
+const UseV2ExchangeModal = ({ onDismiss = defaultOnDismiss, isPageVisit = false }: UseV2ExchangeModalProps) => {
   const [isAcknowledged, setIsAcknowledged] = useState(false)
   const [hasTimerPassed, setHasTimerPassed] = useState(false)
   const [timerSecondsRemaining, setTimerSecondsRemaining] = useState(10)
 
   useEffect(() => {
-    const tick = () => {
-      setTimerSecondsRemaining((prevSeconds) => prevSeconds - 1)
-    }
-
-    const timerInterval = setInterval(() => tick(), 1000)
-
-    const preventClickHandler = (e) => {
-      e.stopPropagation()
-      e.preventDefault()
-      return false
-    }
+    if (isPageVisit) {
+      const tick = () => {
+        setTimerSecondsRemaining((prevSeconds) => prevSeconds - 1)
+      }
 
-    document.querySelectorAll('[role="presentation"]').forEach((el) => {
-      el.addEventListener('click', preventClickHandler, true)
-    })
+      const timerInterval = setInterval(() => tick(), 1000)
 
-    if (timerSecondsRemaining <= 0) {
-      setHasTimerPassed(true)
-      clearInterval(timerInterval)
-    }
+      const preventClickHandler = (e) => {
+        e.stopPropagation()
+        e.preventDefault()
+        return false
+      }
 
-    return () => {
       document.querySelectorAll('[role="presentation"]').forEach((el) => {
-        el.removeEventListener('click', preventClickHandler, true)
+        el.addEventListener('click', preventClickHandler, true)
```

Comment:
Diff is mangled by the `isPageVisit ` condition on ln 41. Nothing has changed within the logic of this useEffect.